src/Microsoft.Azure.SignalR.AspNet/ClientConnections/ClientConnectionManager.cs

```diff
@@ -108,6 +107,49 @@ internal static string GetContentAndDispose(MemoryStream stream)
             }
         }
 
+        private sealed class ClientConnectionHubDispatcher: HubDispatcher
+        {
+            private static readonly char[] SplitChars = new[] { ':' };
+
+            public ClientConnectionHubDispatcher(HubConfiguration config) : base(config)
+            {
+            }
+
+            protected override bool TryGetConnectionId(HostContext context, string connectionToken, out string connectionId, out string message, out int statusCode)
+            {
+                // connectionId is only valid when this method returns true
+                connectionId = null;
+
+                // message and statusCode are only valid when this method returns false
+                message = null;
+                statusCode = 400;
+
+                var tokens = connectionToken.Split(SplitChars, 2);
+
+                connectionId = tokens[0];
+                string tokenUserName = tokens.Length > 1 ? tokens[1] : String.Empty;
+                string userName = GetUserIdentity(context);
+
+                if (!String.Equals(tokenUserName, userName, StringComparison.OrdinalIgnoreCase))
```

Comment:
>String.Equals(tokenUserName, userName, StringComparison.OrdinalIgnoreCase) [](start = 21, length = 74)

could it be possible that invalid token returns String.Empty and GetUserIdentity() also fail to get Identity makes it turn to equal?

Reply:
Update to a simpler one